lib/index.d.ts

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import {
+  StyleProp,
+  TextInputProps,
+  TextProps,
+  ViewProps,
+  TouchableOpacityProps,
+  FlatListProps,
+} from 'react-native';
+
+export interface ToggleProps extends TouchableOpacityProps {
+  onTouch(): void;
+  checked: boolean;
+  iconColor: string;
+}
+
+export interface ItemSelected {
+  id: string | number;
+  item: string;
+}
+
+export class SelectBoxStyleProps {
+  labelStyle?: StyleProp<TextProps>;
+  containerStyle?: StyleProp<ViewProps>;
+  inputFilterContainerStyle?: StyleProp<ViewProps>;
+  inputFilterStyle?: StyleProp<TextInputProps>;
+  optionsLabelStyle?: StyleProp<TextProps>;
+  optionContainerStyle?: StyleProp<ViewProps>;
+  multiOptionContainerStyle?: StyleProp<ViewProps>;
+  multiOptionsLabelStyle?: StyleProp<TextProps>;
+  multiListEmptyLabelStyle?: StyleProp<TextProps>;
+  listEmptyLabelStyle?: StyleProp<TextProps>;
+  selectedItemStyle?: StyleProp<TextProps>;
+}
+
+export class SelectBoxOptionProps extends SelectBoxStyleProps {
+  hideInputFilter?: boolean;
+  isMulti: boolean;
+  removeItemsSelected?: boolean;
+  noEditable?: boolean;
+}
+
+export class SelectBoxVariableProps extends SelectBoxOptionProps {
+  listEmptyText?: string;
+  label: string;
+  inputPlaceholder?: string;
+  width?: number | string;
+  arrowIconColor?: string;
+  searchIconColor?: string;
+  toggleIconColor?: string;
+}
+
+export declare class SelectBoxProps extends SelectBoxVariableProps {
```

Comment:
Hi @hnthangit , i have change group props of SelectBox props. As you mention before. Pls check and confirm.
Thanks

Reply:
This code makes me confuse. Why SelectBoxVariableProps extends SelectBoxOptionProps, 'option' and 'variable' aren't related, keep it small and grouping them in SelectBoxVariableProps. Look like:
`export class SelectBoxStyleProps {
....
}`
`export class SelectBoxOptionProps {
....
}`
`export declare class SelectBoxProps {
    SelectBoxStyleProps,
    SelectBoxOptionProps 
}`